### PR TITLE
adding scroll view and two timer screens

### PR DESCRIPTION
**Pull request by shrevin**

### CODE REVIEW

File: Splash/Splash.xcworkspace/xcuserdata/shreyavinjamuri.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist [hunk 1]
```diff
@@ -685,8 +685,8 @@
             filePath = "Splash/Networking/SpotifyAPIManager.m"
```
Comment (arthpatel29) [inline Splash/Splash.xcworkspace/xcuserdata/shreyavinjamuri.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:685]: This file should not be part of commit. I think we need to add this to ignore file if not already. 

If there is no ignore file, just look up git ignore file and how to add it in the repo.

**Comment by shrevin** [comment]
[](https://drive.google.com/file/d/1XaFlfi_rmkKxcdjJk7C7WNMgEbv-STxb/view?usp=sharing)

File: Splash/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+SpotifyAPIManager.m
```
Comment (arthpatel29) [inline Splash/.gitignore]: Is this intended. This would mean any changes to SpotifyAPIManager would not be included in the commit. 
Example for gitignore file: https://github.com/github/gitignore/blob/main/Objective-C.gitignore

File: Splash/Splash.xcworkspace/xcuserdata/shreyavinjamuri.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist (hunk 1 above)
Comment (arthpatel29) [inline Splash/Splash.xcworkspace/xcuserdata/shreyavinjamuri.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:685]: I think we need to add this to gitignore file as discussed in the previous commit. This is local file, we do not have to commit this.

File: Splash/Splash/Description/DescriptionViewController.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    let redView = UIView(frame: CGRect(x: 20, y: 100, width: 140, height: 100))
+    let animation = CABasicAnimation()
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        redView.backgroundColor = .systemRed
+        view.addSubview(redView)
+        // Do any additional setup after loading the view.
+        animateRight();
+        animateDown()
+    }
```
Comment (yaxuanwang) [inline Splash/Splash/Description/DescriptionViewController.swift]: Are the animation visible?How about moving your animation from `viewDidLoad` to `viewDidAppear`. Usually you are not able to animate the components of your view until it has finished laying out.

File: Splash/Splash/Description/DescriptionViewController.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    
+//    override func viewDidAppear(_ animated: Bool) {
+//        super.viewDidAppear(animated)
+//        let gradientLayer = CAGradientLayer()
+//        gradientLayer.frame = view.bounds
+//        gradientLayer.colors = [UIColor.blue.cgColor,  UIColor.purple.cgColor]
+//        view.layer.addSublayer(gradientLayer)
+//
+//    }
```
Comment (arthpatel29) [inline Splash/Splash/Description/DescriptionViewController.swift]: Let's make sure commented out code is not committed. You can keep it locally. While creating commit just do not include it. I think you can do this using github desktop app.

File: Splash/Splash/TimeWithMusic/TimerXIBView.m [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+//    if ([SpotifyAPIManager shared].appRemote.isConnected) {
+//        [[SpotifyAPIManager shared].appRemote disconnect];
+//    }
+//}
+//
+//- (void)applicationDidBecomeActive:(UIApplication *)application {
+//    if ([SpotifyAPIManager shared].appRemote.connectionParameters.accessToken) {
+//        [[SpotifyAPIManager shared].appRemote connect];
+//    }
+//}
```
Comment (yaxuanwang) [inline Splash/Splash/TimeWithMusic/TimerXIBView.m]: Not needed?

File: Splash/Splash/TimeWithoutMusic/NoMusicViewController.m [hunk 6]
```diff
@@ -33,26 +32,16 @@ @implementation NoMusicViewController
...
-    self.startButton.layer.cornerRadius = 16;
-    self.stopButton.layer.cornerRadius = 16;
-    self.startButton.titleLabel.font = [UIFont fontWithName:@"Bradley Hand Bold" size:19];
-    self.stopButton.titleLabel.font = [UIFont fontWithName:@"Bradley Hand Bold" size:19];
-    dateFormatNoMusic = [[NSDateFormatter alloc] init];
-    dateFormatNoMusic.dateFormat = @"mm:ss";
-    dateFormatNoMusic.timeZone = [NSTimeZone timeZoneWithAbbreviation:@"PST"];
-    self.user = [PFUser currentUser];
-    // layout vc
+
```
Comment (yaxuanwang) [inline Splash/Splash/TimeWithoutMusic/NoMusicViewController.m]: nit: You don't need this method if just calling `[super viewDidLoad];`

File: Splash/Splash/Description/DescriptionViewController.swift (hunk 3 above)
Comment (shrevin) [reply Splash/Splash/Description/DescriptionViewController.swift]: Hi Yaxuan, the animation does show up. I tried putting it in viewDidAppear, however I don't want the animation to happen every time you go back to the screen if you've already opened it.

**Comment by shrevin** [comment]
For the latest commit:

IMG_3879

File: Splash/Splash/TimeWithMusic/TimerXIBView.h [hunk 7]
```diff
@@ -13,6 +13,7 @@ NS_ASSUME_NONNULL_BEGIN
 @interface TimerXIBView : UIView
 - (void) updateTime;
 - (void) updateFontSize:(int)size;
+@property UIViewController *root;
```
Comment (arthpatel29) [inline Splash/Splash/TimeWithMusic/TimerXIBView.h]: This will create a retain cycle when accessed from inside NoMusicViewController. Add 'weak' annotation to make it a weak reference.

File: Splash/Splash/TimeWithoutMusic/NoMusicViewController.m [hunk 8]
```diff
@@ -29,6 +29,10 @@ @implementation NoMusicViewController
 UIAlertController *alertNoMusic;
 bool isPausedNoMusic;
 
+- (void) viewDidLoad {
+    self.stopwatch.root = self;
```
Comment (arthpatel29) [inline Splash/Splash/TimeWithoutMusic/NoMusicViewController.m:33]: As mentioned above, this will create a retain cycle.

File: Splash/Podfile [hunk 9]
```diff
@@ -8,6 +8,7 @@ target 'Splash' do
   # Pods for Splash
   pod 'Parse'
   pod 'Parse/UI'
+  pod 'SWTableViewCell', '~> 0.3.7'
```
Comment (arthpatel29) [inline Splash/Podfile:11]: I think we can get swipe buttons natively in UITableView, you don't have to use a library for that. Search for 'UITableViewRowAction'

File: Splash/Splash/Friends/ViewFriendsViewController.m [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+@implementation ViewFriendsViewController
+NSArray *_rows;
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.tableView.delegate = self;
+    self.tableView.dataSource = self;
+    self.tableView.rowHeight = UITableViewAutomaticDimension;
+    // Do any additional setup after loading the view.        
+        //_testArray = [[NSMutableArray alloc] init];
```
Comment (arthpatel29) [inline Splash/Splash/Friends/ViewFriendsViewController.m]: Let make sure we remove commented out code from commit.

File: Splash/.gitignore [hunk 11]
```diff
@@ -7,23 +7,7 @@ xcuserdata/
...
-*.moved-aside
-*.pbxuser
-!default.pbxuser
-*.mode1v3
-!default.mode1v3
-*.mode2v3
-!default.mode2v3
-*.perspectivev3
-!default.perspectivev3
+Breakpoints_v2.xcbkptlist
```
Comment (arthpatel29) [inline Splash/.gitignore]: Not sure if this will work as intended. 
Giving another reference: https://medium.com/codex/create-git-ignore-file-for-swift-and-xcode-development-d7faaf58893c
